src/complete_hack.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+
+from datetime import datetime, timedelta
+import logging
+import os
+import sys
+import time
+import traceback
+
+import kernelci
+import kernelci.config
+import kernelci.db
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class CompleteHack:
+
+    def __init__(self, configs, args):
+        self._logger = Logger("config/logger.conf", "complete")
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._pending_node = None
+
+    def _check_pending_node(self):
+        now = datetime.utcnow()
+        created = datetime.fromisoformat(self._pending_node['created'])
+        expires = created + timedelta(minutes=3)
```

Comment:
What is the logic for waiting for specifically 3 min after node creation?

Reply:
Tests take about 1min to run on staging with KUnit in Kubernetes, so this is some kind of hard-coded timeout.  Normally all the tests are done within 3min.